analyses/focal-cn-file-preparation/rna-expression-validation.R

```diff
@@ -0,0 +1,175 @@
+
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir)
+}
+
+# Read in data from tsv file (produced in `01-prepare-cn-file.R`)
+cn_df <- readr::read_tsv(file.path(results_dir, "annotated_cn.tsv"))
+
+# Read in RNA-seq expression data
+rsem_expression_polyA <-
+  readr::read_rds(file.path(root_dir,
+                            "data",
+                            "pbta-gene-expression-rsem-fpkm.polya.rds"))
+
+rsem_expression_stranded <-
+  readr::read_rds(file.path(
+    root_dir,
+    "data",
+    "pbta-gene-expression-rsem-fpkm.stranded.rds"
+  ))
+
+# Read in metadata
+metadata <-
+  readr::read_tsv(file.path(root_dir, "data", "pbta-histologies.tsv"))
+
+#### Custom Function -----------------------------------------------------------
+
+merge_expression <-
+  function (copy_number_df, expression_df, metadata, filename){
+    # Given the focal copy number data.frame already annotated with the
+    # metadata, the RNA-seq expression data.frame, and the metadata, combine
+    # the data into one data.frame and save data.frame as tsv file. 
+    #
+    # Args:
+    #   copy_number_df: focal copy number data.frame annotated with the metadata
+    #   expression_df: RNA-seq expression data.frame
+    #   metadata: the relevant metadata data.frame
+    #   filename: filename of the output tsv file
+    #
+    # Returns:
+    #   combined_subset_df: data.frame with information from the focal CN, the
+    #                       RNA-seq expression, and metadata data.frames, which
+    #                       has been subsetted for plotting
+
+    # Change expression data.frame to long tidy format
+    long_expression <- expression_df %>%
+      tidyr::gather(biospecimen_id, expression_value, -gene_id) %>%
+      dplyr::distinct()
+    long_expression$gene_id <-
+      gsub(".*\\_", "", long_expression$gene_id)
+
+    # Annotate expression data with metadata
+    expression_metadata <- long_expression %>%
+      dplyr::inner_join(metadata,
+                        by = c("biospecimen_id" = "Kids_First_Biospecimen_ID")) %>%
+      dplyr::select(
+        biospecimen_id,
+        expression_value,
+        gene_id,
+        Kids_First_Participant_ID,
+        tumor_descriptor
+      ) %>%
+      dplyr::distinct()
+
+    # Merge Focal CN data.frame with RNA expression data.frame
+    combined_df <- copy_number_df %>%
+      dplyr::distinct() %>%
+      dplyr::inner_join(
+        expression_metadata,
+        by = c(
+          "Kids_First_Participant_ID",
+          "gene_symbol" = "gene_id",
+          "tumor_descriptor"
+        )
+      ) %>%
+      dplyr::rename(biospecimen_id_cn = biospecimen_id.x, 
+                    biospecimen_id_expression = biospecimen_id.y) %>%
+      dplyr::mutate(log_expression_value = (log2(expression_value) + 1)) %>%
+      dplyr::mutate(expression_class = dplyr::case_when(
+        expression_value == 0 ~ "0",
+        expression_value > 0 & expression_value <= 10 ~ "0-10",
```

Comment:
How did you choose these values? These feel pretty arbitrary, and it is hard to say what the correct comparative values should be without knowing what the overall distribution is. I think having a few more breaks would not be bad either. Perhaps we want to do these plots with z-scores as well?

Really, we would like to compare this to non-deleted samples, but that might be slow to compute. But if we were to attempt this, it might be nice to see a scatter plot of genes that have deletions, comparing mean expression for samples with deletions to those without.